lib/jsonrpc.ml

```diff
@@ -596,58 +215,63 @@ let version_id_and_call_of_string str =
     | _ -> raise (Malformed_method_request "Invalid request body")
   with Missing_field field ->
     raise (Malformed_method_request (Printf.sprintf "Required field %s is missing" field))
+    | JsonToRpcError json ->
+    raise (Malformed_method_request (Printf.sprintf "Unable to parse %s" (Y.to_string json)))
 
 let call_of_string str =
   let (_, _, call) = version_id_and_call_of_string str in
   call
 
+let get_response extractor str =
```

Comment:
This looks like it has a history. A comment explaining what is going on could be helpful.